dragonfly/implementations/psyllid_provider.py

```diff
@@ -0,0 +1,279 @@
+        return active_channels
+
+
+    def get_number_of_streams(self, channel):
+        stream_count = 0
+        for i in range(3):
+            try:
+                request = '.node-config.ch'+str(i)+'.strw'
+                result = self.provider.get(self.queue_dict[channel]+request)
+                stream_count += 1
+                self.mode_dict[channel]='streaming'
+            except core.exceptions.DriplineError:
+                try:
+                    request = '.node-config.ch'+str(i)+'.trw'
+                    result = self.provider.get(self.queue_dict[channel]+request)
+                    stream_count += 1
+                    self.mode_dict[channel]='triggered'
+                except core.exceptions.DriplineError:
+                    pass  
+        logger.info('Number of streams for channel {}: {}'.format(channel, stream_count))
+        return stream_count
+
+
+    def set_fmt_snr_threshold(self, snr, channel='a'):
+        if self.mode_dict[channel] == 'streaming':
+            logger.warning('Psyllid not in streaming mode')
+            return False
+        request = '.node-config.ch'+str(self.channel_dict[channel])+'.fmt.threshold-power-snr'
+        result = self.provider.set(self.queue_dict[channel]+request, snr)
+        self.deactivate(channel)
+        time.sleep(1)
+        self.activate(channel)
+        return True
+
+
+    def make_trigger_mask(self, channel='a', filename='~/fmt_mask.json'):
+        if self.mode_dict[channel] == 'streaming':
+            return False
+
+        self.request_status(channel)
+        if self.status_value_dict[channel] !=4:
+            self.activate(channel)
+
+        logger.info('freq-only')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.tfrr.freq-only'
+        result = self.provider.cmd(self.queue_dict[channel],request)
+
+        logger.info('update-mask')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.update-mask'
+        result = self.provider.cmd(self.queue_dict[channel],request)
+        time.sleep(1)
+
+        logger.info('run')
+        request = 'start-run'
+        result = self.provider.cmd(self.queue_dict[channel], request)
+        time.sleep(1)
+
+        logger.info('write-mask')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.write-mask'
+        payload = {'filename': filename}
+        result = self.provider.cmd(self.queue_dict[channel], request, payload=payload)
+
+        logger.info('back to time and freq')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.tfrr.time-and-freq'
+        result = self.provider.cmd(self.queue_dict[channel],request)
+
+        logger.info('apply-trigger')
+        request = 'run-daq-cmd.ch'+str(self.channel_dict[channel])+'.fmt.apply-trigger'
+        result = self.provider.cmd(self.queue_dict[channel],request)
+        return True 
```

Comment:
One last comment that applies throughout this file. You capture the result of every provider.cmd in a local variable named `result`. I don't know that the performance hit would ever be noticed, but if you aren't going to then use the content of that variable for anything (usually doing something with the payload values) then there's no need to store it. You can just call the methods and allow the returned value to be garbage collected immediately.